src/commonMain/kotlin/exchange.dydx.abacus/validator/trade/TradeMarketOrderInputValidator.kt

```diff
@@ -79,61 +80,77 @@ internal class TradeMarketOrderInputValidator(
             )
         }
     }
 
-    private fun orderbookSlippage(
+    private fun orderbookOrIndexSlippage(
         trade: Map<String, Any>,
         restricted: Boolean
     ): Map<String, Any>? {
         /*
         MARKET_ORDER_WARNING_ORDERBOOK_SLIPPAGE
         MARKET_ORDER_ERROR_ORDERBOOK_SLIPPAGE
+
+        MARKET_ORDER_WARNING_INDEX_PRICE_SLIPPAGE
+        MARKET_ORDER_ERROR_INDEX_PRICE_SLIPPAGE
          */
-        parser.asNativeMap(trade["summary"])?.let { summary ->
-            parser.asDouble(summary["slippage"])?.let { slippage ->
-                val slippageValue = slippage.abs()
-                if (slippageValue >= MARKET_ORDER_ERROR_SLIPPAGE) {
-                    return error(
-                        if (restricted) "WARNING" else "ERROR",
-                        "MARKET_ORDER_ERROR_ORDERBOOK_SLIPPAGE",
-                        listOf("size.size"),
-                        "APP.TRADE.MODIFY_SIZE_FIELD",
-                        "ERRORS.TRADE_BOX_TITLE.MARKET_ORDER_ERROR_ORDERBOOK_SLIPPAGE",
-                        "ERRORS.TRADE_BOX.MARKET_ORDER_ERROR_ORDERBOOK_SLIPPAGE",
-                        mapOf(
-                            "SLIPPAGE" to mapOf(
-                                "value" to slippageValue,
-                                "format" to "percent",
-                            ),
-                        ),
-                    )
-                } else if (slippageValue >= MARKET_ORDER_WARNING_SLIPPAGE) {
-                    return error(
-                        "WARNING",
-                        "MARKET_ORDER_WARNING_ORDERBOOK_SLIPPAGE",
-                        listOf("size.size"),
-                        null,
-                        "WARNINGS.TRADE_BOX_TITLE.MARKET_ORDER_WARNING_ORDERBOOK_SLIPPAGE",
-                        "WARNINGS.TRADE_BOX.MARKET_ORDER_WARNING_ORDERBOOK_SLIPPAGE",
-                        mapOf(
-                            "SLIPPAGE" to mapOf(
-                                "value" to slippageValue,
-                                "format" to "percent",
-                            ),
-                        ),
-                    )
-                }
-            }
+        val summary = parser.asNativeMap(trade["summary"]) ?: return null
+
+        // missing orderbook slippage (mid price) is most likely due to a one sided liquidity situation
+        // and should be caught by liquidity validation
+        val orderbookSlippage = parser.asDouble(summary["slippage"]) ?: return null
+        val indexSlippage = parser.asDouble(summary["indexSlippage"])
+        val orderbookSlippageValue = orderbookSlippage.abs()
+        val minSlippageValue = min(orderbookSlippageValue, indexSlippage ?: orderbookSlippageValue)
+
+        val isOrderbookSlippageRelevant = indexSlippage == null || orderbookSlippageValue < indexSlippage
+        val slippageType = if (isOrderbookSlippageRelevant) "ORDERBOOK" else "INDEX_PRICE"
```

Comment:
hmm prev 3 lines are all sorta related and maybe hardish to read like this. Maybe consider:

```
var minSlippageValue = orderbookSlippageValue;
var slippageType = "ORDERBOOK"
if (indexSlippage != null && indexSlippage < orderbookSlippageValue) {
 minSlippageValue = indexSlippage
 slippageType = "INDEX_PRICE"
} 
```